### ISSUE TITLE
Exported variable starting with a period are not loaded

### ISSUE THREAD

**Issue by HenrikBengtsson** [tudo-r/BatchJobs#103]
Although `batchExport()` successfully exports variables starting with periods, they are not loaded in the job sessions.  This is in turn because `loadExports()` does not load those variables.

# Example
```r
> library(BatchJobs)
> reg <- makeRegistry(id="test")
> objs <- list(a=1, b=2, .c=3, d=4)
> str(objs)
List of 4
 $ a : num 1
 $ b : num 2
 $ .c: num 3
 $ d : num 4
> batchExport(reg, li=objs)
> keys <- loadExports(reg)
Loading RData files: a,b,d
> keys
[1] "a" "b" "d"
```

# Troubleshooting
All four variables are indeed exported:
```r
> list.files(path=BatchJobs:::getExportDir(reg$file.dir), all.files=TRUE, no..=TRUE)
[1] ".c.RData" "a.RData"  "b.RData"  "d.RData"
```
But [the `ls()` method for fail objects in `loadExports()`](https://github.com/tudo-r/BatchJobs/blob/master/R/Exports.R#L18-L19) uses the default `all.files=FALSE`;
```
> library(fail)
> f = fail(BatchJobs:::getExportDir(reg$file.dir), extension = "RData", simplify=TRUE)
> f$ls()
[1] "a" "b" "d"
> f$ls
function (pattern = NULL)
{
    Ls(.self, pattern)
}
<bytecode: 0x000000000c108dc8>
<environment: 0x000000000c715348>
> fail:::Ls
function (.self, pattern = NULL)
{
    keys = fn2key(.self, list.files(.self$path, pattern = sprintf("\\.%s$",
        .self$extension)))
    if (!is.null(pattern))
        keys = keys[grepl(pattern, keys)]
    return(keys)
}
<bytecode: 0x000000000afff9b8>
<environment: namespace:fail>
```

# Suggestions
* [ ] At a minimum, `batchExport()` should give an error, or at least a warning, if variable names starting with a period are specified, cf. Issue #93.
* [ ] If fixed, the `list.files()` call of `fail:::Ls()` needs `all.files=TRUE` in order to pick up variables starting with a period.  Simple tests, indicates that this is enough, that is, with `.c` in `keys`, then `f$assign(keys, envir = .GlobalEnv)` does indeed load and assign `.c` to the global environment.

**Comment by mllg** [comment]
Should be fixed with fail v1.3 and 7ab2e11b26e511f40be4bbeb4a2bb70e915db983.

**Comment by HenrikBengtsson** [comment]
Thxs.
